File: src/main/java/aquarius0715/commandcountplugin/SQLInsert.java
package aquarius0715.commandcountplugin;

import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

import java.text.SimpleDateFormat;
import java.util.Date;

public class SQLInsert {

    CommandCountPlugin plugin;

    public SQLInsert(CommandCountPlugin plugin) {
        this.plugin = plugin;
    }

    public void insertDefaultTable(Player player) {
        if (!sqlConnectSafely()) {
            return;
        }
        String sql = "INSERT INTO commandCountTable (StartDate, UUID, playerName, cmdCount, scoreBoardStats) VALUE(" + "'"
                + plugin.startDate + "' , '"
                + player.getUniqueId().toString() + "' , '"
                + player.getDisplayName() + "' , "
                + 0 + " , "
                + true + ");";
        plugin.MySQLManager.execute(sql);
    }

    public boolean sqlConnectSafely() {
        if (!plugin.MySQLManager.connectCheck()) {
            Bukkit.broadcastMessage("DB接続に失敗したためプラグインを停止します。");
            plugin.pluginStats = false;
            return false;
        }
        return true;
    }

    public void FormStartTime() {
        Date date = new Date();
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy'-'MM'-'dd' 'HH':'mm':'ss");
        plugin.startDate = simpleDateFormat.format(date);
    }
}

File: src/main/java/aquarius0715/commandcountplugin/SQLInsert.kt
package aquarius0715.commandcountplugin

import org.bukkit.Bukkit
import org.bukkit.entity.Player
import java.util.*

class SQLInsert(var plugin: CommandCountPlugin) : Thread() {
    fun insertDefaultTable(player: Player) {
        Bukkit.getScheduler().runTask(plugin, this)
        run {
            if (!sqlConnectSafely()) {
                return
            }
            val sql = ("INSERT INTO commandCountTable (StartDate, UUID, playerName, cmdCount, scoreBoardStats) VALUE(" + "'"
                    + plugin.StartDate + "' , '"
                    + player.uniqueId.toString() + "' , '"
                    + player.displayName + "' , "
                    + 0 + " , "
                    + true + ");")
            plugin.MySQLManager.execute(sql)
        }
    }

    fun sqlConnectSafely(): Boolean {
        if (!plugin.MySQLManager.connectCheck()) {
            Bukkit.broadcastMessage(plugin.prefix + "DB接続に失敗したためプラグインを停止します。")
            for (player in Bukkit.getOnlinePlayers()) {
                player.sendMessage("レイドが終了しました。")
                if (player.hasPermission("admin")) {
                    plugin.sqlSelect.selectGameResult(player)
                }
                player.scoreboard = Objects.requireNonNull(Bukkit.getScoreboardManager()).newScoreboard
                plugin.StartDate = null
                plugin.gameStats = false
                plugin.playerData.clear()
                plugin.time = -1
                plugin.MySQLManager.close()
            }
            return false
        }
        return true
    }

}